lib/polymorphic_integer_type/activerecord_5_0_0/polymorphic_array_value_extension.rb

```diff
@@ -29,7 +29,7 @@ def type_to_ids_mapping
 
           hash[key] << convert_to_id(value)
         else
-          hash[klass.polymorphic_name] << convert_to_id(value)
+          hash[klass(value).polymorphic_name] << convert_to_id(value)
```

Comment:
The Ruby 3/Rails 7 PR for the original `polymorphic_integer_type` repo called `klass(value)` with the safe navigation operator (ex `klass(value)&.polymorphic_name`), not sure what we risk breaking without it